crud/select/plan.lua

```diff
@@ -79,7 +86,13 @@ local function extract_sharding_key_from_scan_value(scan_value, scan_index, shar
             return nil
         end
 
-        table.insert(sharding_key, field_value)
+        -- check if a field is a part of DDL sharding key
+        if ddl_sharding_key ~= nil and
+           sharding_key_fieldno_map[fieldno] == true then
+            table.insert(sharding_key, field_value)
+        else
+            table.insert(sharding_key, field_value)
```

Comment:
if and else contains the same code...

Reply:
Condition is in a loop and final tables are different.